File: Patterns/05. Right Triangle Patterns (Bottom-Left)/47. Right Triangle BL Binary/blb.java
import java.util.*;
public class blb{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=n; i++){
            for(int j=i; j<=n; j++){ // for(int j=1; j<=(n-i+1); j++)
                System.out.print((j-i+1)%2+ " "); // write each iteration & you know why
            }
            System.out.println();
        }

        sc.close();
    }
}

/*

Use (i+j<=n+1) for optimized code & for n matrix 

Output

1 0 1 0 1 
1 0 1 0 
1 0 1 
1 0 
1  

*/

File: Patterns/05. Right Triangle Patterns (Bottom-Left)/47. Right Triangle BL Binary/blb.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=n; i++){
        for(int j=i; j<=n; j++){ // for(int j=1; j<=(n-i+1); j++)
            cout<< (j-i+1)%2 << " "; // write each iteration & you know why
        }
        cout<<'\n';
    }
}

/*

Use (i+j<=n+1) for optimized code & for n matrix 

Output

1 0 1 0 1 
1 0 1 0 
1 0 1 
1 0 
1   

*/